### ISSUE TITLE
reconstruction layer

### ISSUE THREAD

**Issue by jiangm18** [Cheng-Lin-Li/SegCaps#2]
Hi Cheng

I looked at your code and description. I just want to know why we need reconstruction layer, what if we just use original images and pixel-wised true labels as input and output?

Best

**Comment by Cheng-Lin-Li** [comment]
Hi Jiangm18:

According to the page 5 of the paper states: "As a method of regularization, we extend the idea of reconstructing the input to promote a better embedding of our input space.", it is a regularization technique to prevent over-fitting. 

Kr,
Cheng-Lin

**Comment by jiangm18** [comment]
Hi Cheng
I tried to fit my data with segcaps but it converged too slow and the loss value stuck in -0.58(Dice Coeff), how many epochs you use?

Best
Jiang

**Comment by Cheng-Lin-Li** [comment]
Hi Jiang:

According to my experiments, the model is sensitive on data. You may need to try different loss functions and --recon_wei on your data set. I found the margin loss also works for COCO dataset. I can get a converge result with overfit on single image by 20 epochs (1000 iterations, lr=0.01, recon_wei=20). But I cannot make the model even starting to converge after 20 epochs with 80 color images and enable data augmentation option. Due to the limit GPU resource I have and I believe parameters may need to be adjusted, I stop the experiment so far. According to the url: https://www.kaggle.com/c/data-science-bowl-2018/discussion/54500, author responsed via twitter: SegCaps converged in ~ 232k iters.

Kr,
Cheng-Lin
